src/hob.rs

```diff
@@ -628,8 +628,8 @@ pub struct Capsule {
     /// point to the base of the contiguous memory of the UEFI capsule.
     /// The length of the contiguous memory in bytes.
     ///
-    pub base_address: u8,
-    pub length: u8,
+    pub base_address: u64,

```

Comment:
consider using `EfiPhysicalAddress` for consistency with spec and other structs. it's an alias, so functionally the same, but better self-doc. 
```suggestion
    pub base_address: EfiPhysicalAddress,

```